### PR TITLE
[artwork] Only try embedded artwork if dbmfi has ARTWORK_EMBEDDED flag

### PR DESCRIPTION
**Pull request by chme**
During the `filescanner_ffmpeg` scanning for local files and streams, we already identify which files/streams have embedded artwork. With this PR embedded artwork is only fetched for files/streams that have artwork flag set to ARTWORK_EMBEDDED.

### CODE REVIEW

**Comment by ejurgensen** [comment]
This is brilliant, I had forgotten that flag. With this approach I imagine we could also merge the "embedded" and "embedded http" artwork sources.

I'll hold of merging this a bit, however, because if we with PR #954 decide not to probe the stream at scan time, the embedded flag will probably not be set. There is also the case where the stream is offline at scan time, then the flag is also unset, I suppose. So maybe we should allow for ARTWORK_EMBEDDED and ARTWORK_UNKNOWN? (haven't tested this, so not sure if that will work)

**Comment by chme** [comment]
That we don't fetch embedded artwork for streams that are offline at the time of the scan, is acceptable for me :-)
The performance hit for probing each stream for embedded artwork on the other side, is something i find annoying.

**Comment by ejurgensen** [comment]
While tempting to include this, I'm thinking it might not be the right solution after all. There are two reasons for that, the first is the one I mentioned - I'm still thinking we should remove scan_metadata_stream() from the rssscanner, and then ARTWORK_EMBEDDED won't be possible to use.

The other, and perhaps more important reason, is that the artwork field and the ARTWORK_XXX values aren't otherwise used for anything in forked-daapd. For good reason, perhaps, because the location and source of artwork can always change. So there isn't much value from registering the type of artwork in the db. It could be used for speeding up subsequent searches, i.e. starting with the source indicated by the flag, but I would say we have the cache for that purpose. This field also has some overlap with artwork_url, which is more useful. So what I'm thinking is that the field and the ARTWORK definitions should be removed or deprecated.

We should of course still need to fix the performance issue you highlight. Do you think there is some other way to achieve that? Maybe use artwork_url (not sure how...), or via the cache, i.e. letting it cache the negative result for the entire feed? (I imagine the issue comes from searching for artwork for every podcast in the feed?)

**Comment by chme** [comment]
> is that the artwork field and the ARTWORK_XXX values aren't otherwise used for anything in forked-daapd. For good reason, perhaps, because the location and source of artwork can always change.

The artwork flag only contains the information if the file has embedded artwork. For local files this can only change if the file is changed. And that will trigger forked-daapd to rescan the file and update the artwork flag. So the flag is as correct as the rest of the metadata read from the local file. For remote files (like the podcast episodes) there is a chance, that the file might change and embedded artwork would appear. But this does not sound like a case that happens often. And imo we can neglect. We still have the metadata rescan, that will force a rescan (though admittedly a selective forced rescan would be good to have - i looked into it but got stuck some time ago, and it is not finished/working, hope to look into that sometime again). For the case that the podcast episode file is not available during the scan process, i also don't think we should add special handling for that.

tl;dr i still think this is the right approach, that also benefits searching for artwork for embedded in local files.

> I imagine the issue comes from searching for artwork for every podcast in the feed?

Yes, the problem is, that for podcast "album" artwork, forked-daapd tries for every episode of the podcast to fetch the embedded artwork. The more episodes a podcast has, the longer it takes to search for the artwork. This is done every time a client tries to fetch the artwork for the podcast.

> Maybe use artwork_url (not sure how...), or via the cache, i.e. letting it cache the negative result for the entire feed?

An artwork url field in the files table can be good to have. It would allow to set an artwork url parsed from the feed. But it would not solve the problem for podcasts that do not provide one.

**Comment by ejurgensen** [comment]
> So the flag is as correct as the rest of the metadata read from the local file

Yes, I agree, that is a meaningful and consistent use of the field. I also agree that we can ignore edge cases where the remote media changes. So now I have merged the change in. I have also added a commit that removes the ARTWORK_XXX's that aren't used for anything, and that unlike ARTWORK_EMBEDDED can't be considered file metadata.

Thanks for the extra works, and of course the PR.
